src/Utility/Log.cpp

```diff
@@ -1,59 +1,148 @@
 #include "Log.hpp"
 
-#include <cstdio>
+#include <ostream>
+#include <iostream>
 
 using namespace std;
 
+ostream* Log::streams[NUMBER_OF_CHANNELS];
+
+Log::Log(const Channel channel) {
+    currentChannel = channel;
+}
+
 Log::~Log() {
     fflush(stderr);
 }
 
-Log& Log::operator<<(string text) {
-    fputs(text.c_str(), stderr);
+Log& Log::operator<<(const string& text) {
+    *streams[currentChannel] << text;
+
+#ifdef LOGTESTING
+    if (currentChannel == ERR)
+        throw(text);
+#endif
+
     return *this;
 }
 
-Log& Log::operator <<(int value) {
-    fprintf(stderr, "%i", value);
+Log& Log::operator<<(const int value) {
+    *streams[currentChannel] << value;
+
+#ifdef LOGTESTING
+    if (currentChannel == ERR)
+        throw("Error: i " + std::to_string(value));
+#endif
+
     return *this;
 }
 
-Log& Log::operator <<(unsigned int value) {
-    fprintf(stderr, "%u", value);
+Log& Log::operator<<(const unsigned int value) {
+    *streams[currentChannel] << value;
+
+#ifdef LOGTESTING
+    if (currentChannel == ERR)
+        throw("Error: u " + std::to_string(value));
+#endif
+
     return *this;
 }
 
-Log& Log::operator <<(float value) {
-    fprintf(stderr, "%f", value);
+Log& Log::operator<<(const float value) {
+    *streams[currentChannel] << value;
+
+#ifdef LOGTESTING
+    if (currentChannel == ERR)
+        throw("Error: f " + std::to_string(value));
+#endif
+
     return *this;
 }
 
-Log& Log::operator <<(double value) {
-    fprintf(stderr, "%f", value);
+Log& Log::operator<<(const double value) {
+    *streams[currentChannel] << value;
+
+#ifdef LOGTESTING
+    if (currentChannel == ERR)
+        throw("Error: d " + std::to_string(value));
```

Comment:
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') keyword 'throw' not immediately followed by a semicolon or a single space [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://localhost:9000/coding_rules#rule_key=vera%2B%2B%3AT006)